internal/pkg/convert/node.go

```diff
@@ -214,16 +215,26 @@ func (node *NodeLLB) stepScripts(root llb.State, i int, step v1alpha2.Step) llb.
 		{"test", step.Test},
 	} {
 		for _, instruction := range script.Instructions {
-			root = root.Run(
-				append(node.Graph.commonRunOptions,
-					llb.Args([]string{
-						node.Pkg.Shell.Get(),
-						"-c",
-						instruction.Script(),
-					}),
-					llb.WithCustomName(fmt.Sprintf("%s%s-%d", node.Prefix, script.Desc, i)),
-				)...,
-			).Root()
+			mountOpts := make([]llb.MountOption, 0, 1)
+			runOptions := node.Graph.commonRunOptions
```

Comment:
```suggestion
			runOptions := append([]llb.RunOptions(nil), node.Graph.commonRunOptions...)
```

As `append` might either modify existing slice or re-allocate to new one, we need to protect original value from being changed.